tests/foreman/maintain/test_backup.py

```diff
@@ -0,0 +1,566 @@
+    files = [i for i in files if not re.compile(r'^\.*$').search(i)]
+
+    expected_files = OFFLINE_BACKUP_FILES if backup_type == 'offline' else ONLINE_BACKUP_FILES
+    assert set(files).issuperset(expected_files + CONTENT_FILES), assert_msg
+
+
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_backup_all(sat_maintain, setup_backup_tests, module_synced_repos, backup_type):
+    """Take a backup with all options provided
+
+    :id: bbaf251f-7764-4b7d-b79b-f5f48f5d3b9e
+
+    :steps:
+        1. create an initial backup (for the sake of incremental)
+        2. create another backup with all options provided
+
+    :expectedresult:
+        1. both backups succeed
+    """
+    subdir = f'{BACKUP_DIR}backup-{gen_string("alpha")}'
+    sat_maintain.execute(f'mkdir -m 0777 {subdir}')
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type=backup_type,
+        options={'assumeyes': True},
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    init_backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type=backup_type,
+        options={
+            'assumeyes': True,
+            'force': True,
+            'skip-pulp-content': True,
+            'preserve-directory': True,
+            'split-pulp-tar': '10M',
+            'incremental': init_backup_dir,
+            'features': 'dns,tfp,dhcp,openscap',
+        },
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_backup_offline_logical(sat_maintain, setup_backup_tests, module_synced_repos):
+    """Take an offline backup with '--include-db-dumps' option provided
+
+    :id: dafac83f-75e1-455e-b77f-15fed4eed884
+
+    :steps:
+        1. create a backup
+        2. check that appropriate files are created
+
+    :expectedresult:
+        1. backup succeeds
+        2. files for both, offline and online, backup type are created
+    """
+    subdir = f'{BACKUP_DIR}backup-{gen_string("alpha")}'
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type='offline',
+        options={'assumeyes': True, 'include-db-dumps': True},
+    )
+    logger.info(result.stdout)
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    # Check for expected files
+    backup_dir = re.findall(fr'{subdir}\/satellite-backup-.*-[0-5][0-9]', result.stdout)[0]
+    files = sat_maintain.execute(f'ls -a {backup_dir}').stdout.split('\n')
+    files = [i for i in files if not re.compile(r'^\.*$').search(i)]
+
+    expected_files = OFFLINE_BACKUP_FILES + ONLINE_BACKUP_FILES
+    assert set(files).issuperset(expected_files + CONTENT_FILES), assert_msg
+
```

Comment:
Any specific files we can check for as a result of include-db-dumps?

Reply:
Specific is the `+ ONLINE_BACKUP_FILES` on L541 - the online files are expected too even in case of offline backup.